py2k/serializer.py

```diff
@@ -13,67 +13,58 @@
 # limitations under the License.
 
 
-from uuid import uuid4
+import json
+from typing import List, Dict, Any
 
-from confluent_kafka import SerializingProducer
-from confluent_kafka import KafkaError, Message
+from confluent_kafka.schema_registry import SchemaRegistryClient
+from confluent_kafka.schema_registry.avro import AvroSerializer
+
+from py2k.models import KafkaModel
 
 
 class KafkaSerializer:
-    def __init__(self, topic, key, producer_config):
-        self._topic = topic
+    def __init__(self, item: KafkaModel, key, schema_registry_config: dict):
+        self._item = item
```

Comment:
I agree that we should improve the naming of this, but am not sure about the item -> value. I think the most confusion at this context is:

- key is the field name in item - not a key data to be serialized
- item consist of all the data - value and key to be serialized


The current design of a model and serialization is following:
1. Define data structure you want to push to Kafka - that's the KafkaModel
1. The data structure has no knowledge about what part should be serialized as value and what part as key
1. KafkaSerializer serialize the whole data structure as value
1. When key field is provided, KafkaSerialize will serialize this field as key


Maybe we should let the model know what the key is? (I'm not a fan of this)

Reply:
Why aren't you a fan?

Reply:
- We will have to change the public interface. The key will have to be set during model creation. (but that's not such a problem)
- Not sure how nice would be enforcing key setting for users inheriting from KafkaModel  
    -  It will probably have to be some kind of abstract property

But very happy to take other points of view and reimplement when you guys find it better.

Reply:
Changing the public interface is fine if we have documentation 😋 

But the value schema is a function of the KafkaModel already. It's generated through the Pydantic object. So the Key schema should be there too IMO.

Reply:
Yop, makes sense. I'll try to put it together today. When it shows up to be more work than expected, we can move it to next PR

Reply:
I think it might be a pretty big change. I'm happy to merge this now and have a different PR for this

Reply:
> maybe it's worth calling this `value`? i.e. we have some key, value serialization? Not sure.
> 
> @felipemmelo thoughts?

In Kafka parlance, the stuff it received is called a `record`, which is composed of `key` and `value`, so might be worth it to follow their naming strategies.

Reply:
> Changing the public interface is fine if we have documentation 😋
> 
> But the value schema is a function of the KafkaModel already. It's generated through the Pydantic object. So the Key schema should be there too IMO.

and examples, imo